leapp/cli/upgrade/__init__.py

```diff
@@ -175,6 +173,52 @@ def upgrade(args):
         sys.exit(1)
 
 
+@command('preupgrade', help='Generate preupgrade report')
+@command_opt('whitelist-experimental', action='append', metavar='ActorName',
+             help='Enables experimental actors')
+def preupgrade(args):
+    if os.getuid():
+        raise CommandError('This command has to be run under the root user.')
+
+    if args.whitelist_experimental:
+        args.whitelist_experimental = list(itertools.chain(*[i.split(',') for i in args.whitelist_experimental]))
+    context = str(uuid.uuid4())
+    configuration = {
+        'debug': os.getenv('LEAPP_DEBUG', '0'),
+        'verbose': os.getenv('LEAPP_VERBOSE', '0'),
+        'whitelist_experimental': args.whitelist_experimental or ()
+    }
+    e = Execution(context=context, kind='preupgrade', configuration=configuration)
+    e.store()
+    archive_logfiles()
+    logger = configure_logger('leapp-preupgrade.log')
+    os.environ['LEAPP_EXECUTION_ID'] = context
+
+    try:
+        repositories = load_repositories()
+    except LeappError as exc:
+        raise CommandError(exc.message)
+    workflow = repositories.lookup_workflow('IPUWorkflow')()
+    for actor_name in configuration.get('whitelist_experimental', ()):
+        actor = repositories.lookup_actor(actor_name)
+        if actor:
+            workflow.whitelist_experimental_actor(actor)
+        else:
+            msg = 'No such Actor: {}'.format(actor_name)
+            logger.error(msg)
+            raise CommandError(msg)
+    with beautify_actor_exception():
+        until_phase = 'ReportsPhase'
+        logger.info('Executing workflow until phase: %s', until_phase)
+        workflow.run(context=context, until_phase=until_phase)
+
+    report_errors(workflow.errors)
+    report_report('/var/log/leapp/leapp-report.txt')
```

Comment:
Please do not make leapp-report path a magic constant

Reply:
It's in the config now.